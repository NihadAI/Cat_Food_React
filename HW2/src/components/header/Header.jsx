import PropTypes from 'prop-types';
import Cart from '../cart';
import styles from './Header.module.css';
import Favourites from '../favourites';

function Header({
	cartCounter,
	favCounter,
	setInCartProducts,
	setInFavProducts,
}) {
	return (
		<header className={styles.header}>
			<div className={styles.container}>
				<a
					href='/'
					className={styles.logo}>
					CatShop
				</a>
				<div className={styles.wrapper}>
					<Cart
						counter={cartCounter}
						setInCartProducts={setInCartProducts}
					/>
					<Favourites
						counter={favCounter}
						setInFavProducts={setInFavProducts}
					/>
				</div>
			</div>
		</header>
	);
}

Header.propTypes = {
	cartCounter: PropTypes.number,
	favCounter: PropTypes.number,
	setInCartProducts: PropTypes.func,
	setInFavProducts: PropTypes.func,
};

Header.defaultProps = {
	cartCounter: 0,
	favCounter: 0,
	setInCartProducts: () => {},
	setInFavProducts: () => {},
};

export default Header;
