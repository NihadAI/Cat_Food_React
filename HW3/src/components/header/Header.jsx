import PropTypes from 'prop-types';
import Cart from '../cart';
import styles from './Header.module.css';
import Favourites from '../favourites';
import { Link } from 'react-router-dom';

function Header({ cartCounter, favCounter }) {
	return (
		<header className={styles.header}>
			<div className={styles.container}>
				<Link
					to='/'
					className={styles.logo}>
					CatShop
				</Link>

				<div className={styles.wrapper}>
					<Link to='/cart'>
						<Cart counter={cartCounter} />
					</Link>
					<Link to='/favourites'>
						<Favourites counter={favCounter} />
					</Link>
				</div>
			</div>
		</header>
	);
}

Header.propTypes = {
	cartCounter: PropTypes.number,
	favCounter: PropTypes.number,
	setInCartProducts: PropTypes.func,
	setInFavProducts: PropTypes.func,
};

Header.defaultProps = {
	cartCounter: 0,
	favCounter: 0,
	setInCartProducts: () => {},
	setInFavProducts: () => {},
};

export default Header;
