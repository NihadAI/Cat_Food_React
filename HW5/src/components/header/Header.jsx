import styles from './Header.module.css';
import { NavLink } from 'react-router-dom';
import CartCounter from '../cartCounter';
import FavouritesCounter from '../favouritesCounter';

function Header() {
	return (
		<header className={styles.header}>
			<div className={styles.container}>
				<NavLink
					to='/'
					className={styles.logo}>
					CatShop
				</NavLink>

				<div className={styles.wrapper}>
					<NavLink to='/cart'>
						<CartCounter />
					</NavLink>
					<NavLink to='/favourites'>
						<FavouritesCounter />
					</NavLink>
				</div>
			</div>
		</header>
	);
}

export default Header;
