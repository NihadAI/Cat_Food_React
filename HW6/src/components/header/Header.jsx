import styles from './Header.module.css';
import { NavLink, useLocation } from 'react-router-dom';
import CartCounter from '../cartCounter';
import FavouritesCounter from '../favouritesCounter';
import { useContext } from 'react';
import ProductsViewContext from '../../context/ProductsView';

function Header() {
	const { view, toggleView } = useContext(ProductsViewContext);
	const { pathname } = useLocation();

	return (
		<header className={styles.header}>
			<div className={styles.container}>
				<NavLink
					to='/'
					className={styles.logo}>
					CatShop
				</NavLink>

				{(pathname === '/' || pathname === '/favourites') && (
					<div className={styles.controls}>
						<button
							className={styles.switcher}
							onClick={toggleView}>
							{view} view
						</button>
					</div>
				)}

				<div className={styles.wrapper}>
					<NavLink to='/cart'>
						<CartCounter />
					</NavLink>
					<NavLink to='/favourites'>
						<FavouritesCounter />
					</NavLink>
				</div>
			</div>
		</header>
	);
}

export default Header;
