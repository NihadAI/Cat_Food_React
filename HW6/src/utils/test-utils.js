import React from 'react';
import { render as rtlRender, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { MemoryRouter } from 'react-router-dom';

import currentsReducer from '../redux/slices/currentsSlice';
import cartReducer from '../redux/slices/cartSlice';
import favouritesReducer from '../redux/slices/favouritesSlice';
import modalReducer from '../redux/slices/modalSlice';
import itemsReducer from '../redux/slices/itemsSlice';

function render(ui, options = {}) {
	const initialState = {
		currents: { defaultScreenText: 'Test Screen Text' },
		modal: {
			isModalOpen: false,
			modalAction: {
				title: 'Test Title',
				content: 'Test Content',
				button: <button>Test button</button>,
			},
			isShadowDisabled: false,
		},
		cart: { items: [], count: 0, fullPrice: 0 },
		favourites: { items: [], count: 0 },
		items: { items: [], isLoading: true },
	};

	screen.debug();

	const testStore = configureStore({
		reducer: {
			currents: currentsReducer,
			cart: cartReducer,
			favourites: favouritesReducer,
			modal: modalReducer,
			items: itemsReducer,
		},
		preloadedState: initialState,
	});

	const Wrapper = ({ children }) => (
		<MemoryRouter initialEntries={['/test-route']}>
			<Provider store={testStore}>{children}</Provider>
		</MemoryRouter>
	);

	return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export { render, screen };
