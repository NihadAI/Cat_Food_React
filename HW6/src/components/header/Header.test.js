import { render } from '../../utils/test-utils';
import Header from '.';

describe('Smoke test', () => {
	it('should render', () => {
		const { asFragment } = render(<Header />);

		expect(asFragment()).toMatchSnapshot();
	});
});
