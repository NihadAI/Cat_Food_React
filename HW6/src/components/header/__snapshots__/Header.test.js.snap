// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke test should render 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <a
        class="logo"
        href="/"
      >
        CatShop
      </a>
      <div
        class="wrapper"
      >
        <a
          class=""
          href="/cart"
        >
          <div
            class="cart"
          >
            Cart 
            <span
              class="counter"
            >
              0
            </span>
          </div>
        </a>
        <a
          class=""
          href="/favourites"
        >
          <div
            class="fav"
          >
            Favourites 
            <span
              class="counter"
            >
              0
            </span>
          </div>
        </a>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
